sklearn/grid_search.py

```diff
@@ -398,6 +398,11 @@ def __init__(self, estimator, scoring=None,
         self.pre_dispatch = pre_dispatch
         self.error_score = error_score
 
+        if not self.iid:
```

Comment:
iid should be set to None and if it's not none, the warning should be shown, and if it is None, it should behave as ``True`` (as that's the current behavior)